### PR TITLE
SDS-14449: Fixed access to headers' variable with wrong key

### LINKED ISSUE #86: ReferenceEntityMediaFileApi cannot recognise code in response
Hi All, 

The method _ReferenceEntityMediaFileApi::extractCodeFromCreationResponse_ checks the incorrect header key here:
`https://github.com/akeneo/api-php-client-ee/blob/1577caa4ee0f94749d331cc6a225250620cd568c/src/Api/ReferenceEntityMediaFileApi.php#L76`
Instead of the `$headers['Reference-entities-media-file-code'] `it should be `$headers['Reference-Entities-Media-File-Code']`

Without this fix it is impossible to create a media file for the reference entity. Could you please adjust it? 

Thank you.

### LINKED ISSUE #93: Wrong asset media file code response header
`AssetMediaFileApi::extractCodeFromCreationResponse()` looks for a response header `Asset-media-file-code` to extract a code of the newly-created asset media file. PIM doesn't return `Asset-media-file-code`, it returns `Asset-Media-File-Code` (different capitalisation). So the client concludes this header doesn't exist and throws an exception.

Here's a proof that PIM returns a different header:
Screenshot 2020-07-12 at 19 37 39

This is a successful asset media POST request made to Akeneo Serenity edition. As you can see, the response is 201, but the header is different. I don't know why, I checked the source code of Akeneo EE and found that actually `Asset-media-file-code` is returned. So I'm not sure what's wrong here.

Maybe it would be a good idea to check for the header in a case-insensitive way (using `strcasecmp()`) or just check both `Asset-media-file-code` and `Asset-Media-File-Code`.

### PR DESCRIPTION
**Pull request by GiulianoDiVita**

### CODE REVIEW

File: src/Api/AssetManager/AssetMediaFileApi.php [hunk 1]
```diff
@@ -67,10 +67,10 @@ private function extractCodeFromCreationResponse(ResponseInterface $response): s
     {
         $headers = $response->getHeaders();
 
-        if (!isset($headers['Asset-media-file-code'][0])) {
```
Comment (mmetayer) [inline src/Api/AssetManager/AssetMediaFileApi.php:70]: Actually HTTP headers are case insensitive according to [RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2). We cannot guarantee that this case will be respected (the server can respond with either of `Asset-Media-File-Code`. `asset-media-file-code`, `Asset-media-file-code`, ... headers)

**Review by mmetayer — CHANGES_REQUESTED** [review]

**Comment by mmetayer** [comment]
Hey @GiulianoDiVita, I was about to submit a PR myself when I saw yours. As I explained in my review, I don't think your solution is correct, a better solution would be IMO:
```php
    private function extractCodeFromCreationResponse(ResponseInterface $response): string
    {
        $header = $response->getHeader('Asset-media-file-code'); // this method is case-insensitive, and returns an empty array if the header is not set

        if (empty($header)) {
            throw new RuntimeException('The response does not contain the code of the created media-file.');
        }

        return (string) $header[0];
    }
```

Would youlike me to submit a PR or would you rather fix yours?

**Comment by GiulianoDiVita** [comment]
> Hey @GiulianoDiVita, I was about to submit a PR myself when I saw yours. As I explained in my review, I don't think your solution is correct, a better solution would be IMO:
> 
> ```
>     private function extractCodeFromCreationResponse(ResponseInterface $response): string
>     {
>         $header = $response->getHeader('Asset-media-file-code'); // this method is case-insensitive, and returns an empty array if the header is not set
> 
>         if (empty($header)) {
>             throw new RuntimeException('The response does not contain the code of the created media-file.');
>         }
> 
>         return (string) $header[0];
>     }
> ```
> 
> Would youlike me to submit a PR or would you rather fix yours?

Hi @mmetayer,

I modified the code following your review, hope now is ok.

Thank you,
Giuliano

**Comment by mmetayer** [comment]
Nice!

I took the liberty of adding some tests and rebasing on master, I hope you don't mind :slightly_smiling_face:

**Review by mmetayer — APPROVED** [review]

**Comment by jbieliauskas** [comment]
Wait, I'm sorry, how does this solve the problem if a header is `Asset-media-file-code` (like in [documentation](https://api.akeneo.com/api-reference.html#post_asset_media_files)) or `Asset-Media-File-Code` (like in this [issue](#93))?

**Comment by mmetayer** [comment]
@justasbieliauskas the HTTP header keys are case-insensitive (see [RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2)). Theoretically the HTTP server can include the header with any case
The issue was that we tested its existence with PHP's native `isset` function (which _is_ case sensitive).
By using the MessageInterface::getHeader() message, we ensure getting the correct header, regardless of its case (see PHPDoc [here](https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/MessageInterface.php#L94))
To sum up, ```$response->getHeader('Asset-media-file-code')``` will return any of the following headers:
- Asset-media-file-code
- Asset-Media-File-Code
- asset-media-file-code
- ...

File: src/Api/AssetManager/AssetMediaFileApi.php [hunk 2]
```diff
@@ -65,12 +65,12 @@ public function create($mediaFile): string
      */
     private function extractCodeFromCreationResponse(ResponseInterface $response): string
     {
-        $headers = $response->getHeaders();
+        $header = $response->getHeader('asset-media-file-code');
```
Comment (SamirBoulil) [inline src/Api/AssetManager/AssetMediaFileApi.php:68]:
```suggestion
        $assetMediaFileCode = $response->getHeader('asset-media-file-code');
```

**Review by SamirBoulil — APPROVED** [review]
